gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageTest.java

```diff
@@ -416,6 +417,70 @@ public void upload_failure_runtimeException() throws Exception {
         .inOrder();
   }
 
+  @Test
+  public void upload_retry_requestTimeout() throws Exception {
+    byte[] testData = new byte[MediaHttpUploader.MINIMUM_CHUNK_SIZE];
```

Comment:
let's add a test for multiple chunks

Reply:
RetryHttpInitializer is used for all the requests made for Json APIs. I think this test should be enough to display the retry behavior.